File: java/src/nz/sodium/time/SecondsTimerSystem.java
package nz.sodium.time;

/**
 * A timer system implementation where the clock is a floating point number of seconds
 * since program start.
 */
public class SecondsTimerSystem extends TimerSystem<Double> {
    public SecondsTimerSystem() {
        super(new SecondsTimerSystemImpl());
    }
}

File: dotnet/src/Sodium/CSharp/Sodium.Frp/Time/SecondsTimerSystem.cs
using System;

namespace Sodium.Frp.Time
{
    /// <summary>
    ///     A timer system using the number of seconds since the application started.
    /// </summary>
    public sealed class SecondsTimerSystem : TimerSystem<double>
    {
        public SecondsTimerSystem(Action<Exception> handleException)
            : base(new Implementation(), handleException)
        {
        }

        private class Implementation : TimerSystemImplementationImplementationBase<double>
        {
            private readonly DateTime startTime;

            public Implementation() => this.startTime = DateTime.Now;

            protected override TimeSpan SubtractTimes(double first, double second) =>
                TimeSpan.FromSeconds(first - second);

            public override double Now => (DateTime.Now - this.startTime).TotalSeconds;
        }
    }
}